tests/dragonfly/snapshot_test.py

```diff
@@ -583,38 +583,25 @@ async def test_big_value_serialization_memory_limit(df_factory, query):
     instance.start()
     client = instance.client()
 
-    ten_mb = 10_000_000
+    one_gb = 1_000_000_000
+    elements = 1000
+    one_mb = 1_000_000
 
-    def ten_mb_random_string():
-        return "".join(random.choices(string.ascii_letters, k=ten_mb))
-
-    one_gb = 1_000_000_000  # 1GB
-
-    upper_limit = one_gb * 1.1  # 1GB + 100MB
-
-    i = 0
-
-    while instance.rss < one_gb:
-        if query == "HSET":
-            i = i + 1
-            await client.execute_command(f"HSET foo_key foo_{i} {ten_mb_random_string()}")
-        elif query == "SADD":
-            await client.execute_command(f"SADD foo_key {ten_mb_random_string()}")
-        elif query == "ZSET":
-            await client.execute_command(f"ZADD foo_key {i} {ten_mb_random_string()}")
-        elif query == "LIST":
-            await client.execute_command(f"LPUSH foo_key {ten_mb_random_string()}")
-
-    async def check_memory_usage(instance):
-        while True:
-            assert instance.rss < upper_limit
-            await asyncio.sleep(0.01)
-
-    checker = asyncio.create_task(check_memory_usage(instance))
+    await client.execute_command(
```

Comment:
I would add a comment here what it exectly does.
i.e add 1 db entry of given type with elements num each one of size element size

Now my question is why do we end up with more than 2g rss if we have one entry of 1000 elemets each is 1Mb size

Reply:
RSS will grow during DEBUG populate, For example run:

```
debug populate 1 prefix 1000000 TYPE hash RAND ELEMENTS 1000
```

Creates a hash table with 1GB elements. Now do `INFO MEMORY`. RSS is: 2062901248 (2GB ~ doubled).

Reply:
This does not answer the question why, but we can continue with this PR and please create another github ticket so we can follow up on this to investigate why we have this overhead

Reply:
uh you meant what is causing the rss spike. I do not know, it was an observation. I created an issue https://github.com/dragonflydb/dragonfly/issues/4124